src/main/java/pl/lodz/p/it/zzpj/botsite/services/BotServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package pl.lodz.p.it.zzpj.botsite.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pl.lodz.p.it.zzpj.botsite.entities.Bot;
+import pl.lodz.p.it.zzpj.botsite.exceptions.*;
+import pl.lodz.p.it.zzpj.botsite.repositories.BotRepository;
+
+import java.util.Optional;
+
+@Service("mongoBotService")
+public class BotServiceImpl implements BotService {
+    private final BotRepository botRepository;
+
+    @Autowired
+    public BotServiceImpl(BotRepository botRepository) {
+        this.botRepository = botRepository;
+    }
+
+    public Bot findById(String id) throws BotRetrievalException {
```

Comment:
Nie jest to wymagane ze strony jezyka, ale dodawaj @Override nad metodami nadpisanymi - ułatwia nieco czytanie i rozóżnienie.